walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,82 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.LinkedList;
+import java.util.Comparator;
 
 public final class FindMeetingQuery {
+  private List<TimeRange> busyTimes = new ArrayList<>();
+  private List<TimeRange> freeTimes = new ArrayList<>();
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    busyTimes = getBusyTimes(events, request);
+    return (getFreeTimes(findOverlappingTimes(), request));
+  }
+
+  // Compare attendees of given events with attendees for requested meeting to determine conflicts
+  private List<TimeRange> getBusyTimes(Collection<Event> events, MeetingRequest request) {
+
+    Collection<String> attendees = request.getAttendees();
+    for (Event event : events) {
+      if(!(Collections.disjoint(event.getAttendees(), attendees))) {
+        busyTimes.add(event.getWhen());
+      }
+    }
+    Collections.sort(busyTimes, TimeRange.ORDER_BY_START);
+    return busyTimes;
+  }
+
+  // Returns list of consolidated busy time ranges accounting for overlapping events
+  private List<TimeRange> findOverlappingTimes() {
+    List<TimeRange> overlappingTimes = new ArrayList<>();
+    // variables used to keep track of start and end times of previous relevant busy time blocks
+    int prevStartTime = -1;
+    int prevEndTime = 0;
+    for (TimeRange timeRange : busyTimes) {
+      if (prevStartTime == -1) {
+        prevStartTime = timeRange.start();
+        prevEndTime = timeRange.end();
+      }
+      // events overlap
+      else if ((timeRange.start() <= prevEndTime) && (timeRange.end() >= prevEndTime)){
+        prevEndTime = timeRange.end();
+      }
```

Comment:
Won't this always be true on the second iteration of this loop, assuming there's more than one element in busyTimes?